ui/admin/app/components/form/target/index.hbs

```diff
@@ -1,122 +1,6 @@
-  >
-    {{#if @model.errors.name}}
-      <field.errors as |errors|>
-        {{#each @model.errors.name as |error|}}
-          <errors.message>{{error.message}}</errors.message>
-        {{/each}}
-      </field.errors>
-    {{/if}}
-  </form.input>
-
-  <form.textarea
-    @name='description'
-    @type='text'
-    @value={{@model.description}}
-    @label={{t 'form.description.label'}}
-    @helperText={{t 'form.description.help'}}
-    @error={{@model.errors.description}}
-    as |field|
-  >
-    {{#if @model.errors.description}}
-      <field.errors as |errors|>
-        {{#each @model.errors.description as |error|}}
-          <errors.message>{{error.message}}</errors.message>
-        {{/each}}
-      </field.errors>
-    {{/if}}
-  </form.textarea>
-
-  <form.input
-    @name='session_max_seconds'
-    @type='number'
-    @value={{@model.session_max_seconds}}
-    @label={{t 'form.session_max_seconds.label'}}
-    @error={{@model.errors.session_max_seconds}}
-    @helperText={{t 'form.session_max_seconds.help'}}
-    min='0'
-    step='1'
-    as |field|
-  >
-    {{#if @model.errors.session_max_seconds}}
-      <field.errors as |errors|>
-        {{#each @model.errors.session_max_seconds as |error|}}
-          <errors.message>{{error.message}}</errors.message>
-        {{/each}}
-      </field.errors>
-    {{/if}}
-  </form.input>
-
-  <form.input
-    @name='session_connection_limit'
-    @type='number'
-    @value={{@model.session_connection_limit}}
-    @label={{t 'form.session_connection_limit.label'}}
-    @error={{@model.errors.session_connection_limit}}
-    @helperText={{t 'form.session_connection_limit.help'}}
-    min='-1'
-    step='1'
-    as |field|
-  >
-    {{#if @model.errors.session_connection_limit}}
-      <field.errors as |errors|>
-        {{#each @model.errors.session_connection_limit as |error|}}
-          <errors.message>{{error.message}}</errors.message>
-        {{/each}}
-      </field.errors>
-    {{/if}}
-  </form.input>
-
-  {{! TODO this is a type-specific field }}
-  <form.input
-    @name='default_port'
-    @type='number'
-    @value={{@model.default_port}}
-    @label={{t 'form.default_port.label'}}
-    @error={{@model.errors.default_port}}
-    @helperText={{t 'form.default_port.help'}}
-    min='0'
-    step='1'
-    as |field|
-  >
-    {{! TODO need to figure out how to reference `attributes.errors` here }}
-    {{#if @model.errors.default_port}}
-      <field.errors as |errors|>
-        {{#each @model.errors.default_port as |error|}}
-          <errors.message>{{error.message}}</errors.message>
-        {{/each}}
-      </field.errors>
-    {{/if}}
-  </form.input>
-  {{#if (can 'save model' @model)}}
-    <form.actions
-      @disabled={{if @model.cannotSave @model.cannotSave}}
-      @enableEditText={{t 'actions.edit-form'}}
-      @submitText={{t 'actions.save'}}
-      @cancelText={{t 'actions.cancel'}}
-    />
-  {{/if}}
-</Rose::Form>
\ No newline at end of file
+{{component
```

Comment:
Asking without context:

I see `targets/ssh/index.hbs` and `targets/tcp/index.hbs` being the same except for the reference to the translation within the label and helper text in example: `resources.target.types.ssh`.

As per my understanding both `ssh` and `tcp` have the exact same fields?

My question is, due to that small difference mentioned above, do we want to use difference templates or we can use one template file?

Reply:
We have been distinguishing by type.  I'm not opposed to sharing a single component if they are the same, but they should still be separated from the "index" component, since future types may indeed be different.  The problem might be naming.  What would we call it?  It's not a "base" because future types that are in fact different aren't really a subclass or child.  Something to think about.

Reply:
Also wondering if it should be an additional layer to achieve simplicity in the index component.  e.g. index -> ssh -> "generic" (rather than index -> "generic")

Reply:
I am a fan of "generic" and I like the index -> ssh -> approach. My assumption is we would use this component in the `targets/ssh/index.hbs` and just pass the more dynamic parts like translations.

Reply:
I went with `details.hbs` and followed the suggested index->ssh->details. feedback is much appreciated!